### PR TITLE
Add AvailabilityStatus to Applications, Authentications, and Endpoints

### PR DESCRIPTION
**Pull request by agrare**
Add AvailabilityStatus to Applications, Authentications, and Endpoints.

Extend the granularity of availability_status by allowing more resources
to contain availability status and an error message.

### CODE REVIEW

File: app/models/application.rb [hunk 1]
```diff
@@ -2,4 +2,7 @@ class Application < ApplicationRecord
   include TenancyConcern
   belongs_to :source
   belongs_to :application_type
+
+  attribute :availability_status, :string
```
Comment (bdunne) [inline app/models/application.rb]: Any reason we need to set the `cast_type` on these strings?

Comment (agrare) [reply app/models/application.rb]: I copied what was already in [Source](https://github.com/ManageIQ/sources-api/blob/master/app/models/source.rb#L9)
We can change it but would like to keep it consistent across the board

**Comment by slemrmartin** [comment]
@agrare what is the use case of AvailabilityStatus for Application? Either the Source is unavailable or Application is unavailable, but I cannot imagine why Application will be unavailable for one Source

**Comment by agrare** [comment]
@slemrmartin `Application` is the link between a `Source` and an `ApplicationType` like `catalog` or `cost-management`.

`availability_status` on an application can indicate for example that the ARN credentials for cost-management are fine, but the S3 bucket (app specific) is missing.

**Pull request merged**